nixos/modules/tasks/network-interfaces.nix

```diff
@@ -924,6 +924,14 @@ in
       (flip map interfaces (i: {
         assertion = i.subnetMask == null;
         message = "The networking.interfaces.${i.name}.subnetMask option is defunct. Use prefixLength instead.";
+      })) ++ (flip map interfaces (i: {
+        # With the linux kernel, interface name length is limited by IFNAMSIZ
+        # to 16 bytes, including the trailing null byte.
+        # See include/linux/if.h in the kernel sources
+        assertion = stringLength i.name < 16;
+        message = ''
+          The name of networking.interfaces."${i.name}" is too long, it needs to be less than 16 characters.
```

Comment:
Why not adhere to the surrounding code and say just `${i.name}` instead of adding `"`s?

Reply:
@4z3 The idea was to avoid misinterpretations when the interface name contains a dot.

e.g. the tinc nixos module names interfaces like `tinc.${networkname}`, which would result in the error message `The name of networking.interfaces.tinc.examplenet is too long, it needs to be less than 16 characters`. This would obscure that the limit is not broken by `${networkname}`, but by `tinc.${networkname}` (and also look like `tinc` is a submodule of `networking.interfaces`).

But yes, this is now inconsistent with the surrounding code, forgot about this. What do you think?

Reply:
Valid point. Maybe the surroundings should be updated then, for consistency and also because they can produce misinterpretable error messages.

Reply:
Ok! Fix at https://github.com/NixOS/nixpkgs/pull/31476